src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -1,278 +1,477 @@
-                "SystemUsesLightTheme", null) as int?;
+            bool useLightColors = GetUseLightColors(Application.Current.ThemeMode);
+            window.SetImmersiveDarkMode(!useLightColors);
+            WindowBackdropManager.SetBackdrop(window, WindowBackdropType.MainWindow);
+        }
+        else
+        {
+            // TODO : Remove the styles from windows which have BackdropDisabledWidowStyle
+            window.SetImmersiveDarkMode(false);
+            WindowBackdropManager.SetBackdrop(window, WindowBackdropType.None);
         }
 
-        return useLightTheme != null && useLightTheme != 0;
     }
 
-    /// <summary>
-    ///  Update the Fluent theme resources with the values in new dictionary.
-    /// </summary>
-    /// <param name="dictionaryUri"></param>
-    private static void AddOrUpdateThemeResources(Uri dictionaryUri)
+    private static void ApplyStyleOnWindow(Window window, bool useLightColors)
     {
-        ArgumentNullException.ThrowIfNull(dictionaryUri, nameof(dictionaryUri));
-
-        var newDictionary = new ResourceDictionary() { Source = dictionaryUri };
+        if(window == null || window.IsDisposed) return;
 
-        FindFluentThemeResourceDictionary(out ResourceDictionary fluentDictionary);
-        
-        if (fluentDictionary != null)
+        // We only apply Style on window, if the Window.Style has not already been set to avoid overriding users setting. 
+        if(window.Style == null)
         {
-            Application.Current.Resources.MergedDictionaries.Remove(fluentDictionary);
+            window.SetResourceReference(FrameworkElement.StyleProperty, typeof(Window));
         }
 
-        Application.Current.Resources.MergedDictionaries.Add(newDictionary);
+        window.SetImmersiveDarkMode(!useLightColors);
+
+        if(SystemParameters.HighContrast)
+        {
+            WindowBackdropManager.SetBackdrop(window, WindowBackdropType.None);
+        }
+        else
+        {
+            WindowBackdropManager.SetBackdrop(window, WindowBackdropType.MainWindow);
+        }
     }
 
     #endregion
 
-    #endregion
 
     #region Internal Properties
 
-    internal static bool IsFluentThemeEnabled => _isFluentThemeEnabled;
-    // TODO : Find a better way to deal with different default font sizes for different themes.
+    internal static bool DeferSyncingThemeModeAndResources { get; set; } = true;
+
+    internal static bool IsFluentThemeEnabled
+    {
+        get
+        {
+            if(Application.Current == null) return false;
+            return Application.Current.ThemeMode != ThemeMode.None;
+        }
+    }
+
+    internal static bool DeferredAppThemeLoading { get; set; } = false;
+
+    internal static bool IgnoreAppResourcesChange { get; set; } = false;
+
     internal static double DefaultFluentThemeFontSize => 14;
 
+    internal static WindowCollection FluentEnabledWindows { get; set; } = new WindowCollection();
```

Comment:
This will hold all windows in memory forever that ever had a fluent theme applied to them, doesn't it?

Reply:
Resolved in PR #9514